### PR TITLE
HGWGDC-21161: Logging of unauthorised requests

### PR DESCRIPTION
**Pull request by chaitanyasingla-dt**

### CODE REVIEW

File: forwarder.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ func forwarder(c echo.Context, client *http.Client) error {
 	req := c.Request()
 	ctx := req.Context()
 
+	isAuthHeaderCheckEnabled := viper.GetBool("authHeaderCheck.enable")
```
Comment (Sachin4403) [inline forwarder.go]: can you read it once and use it further?

File: petasos-rewriter.yaml [hunk 2]
```diff
@@ -85,4 +85,9 @@ metricsOptions:
   namespace: xmidt
   # subsystem is the subsystem of the metrics provided
   # (Optional)
-  subsystem: petasos_rewriter
\ No newline at end of file
+  subsystem: petasos_rewriter
+
+# If enabled, requests redirected to Talaria will be examined for the presence of an Authorization header, 
+# and if the Authorization header is not present, the request will be rejected.
+authHeaderCheck:
```
Comment (Sachin4403) [inline petasos-rewriter.yaml]: make it like  server.authheader.check.enabled.
 server.authheader.check.requestPath:

File: metrics.go [hunk 3]
```diff
@@ -73,17 +83,42 @@ func registerMetrics() *metricRegistry {
...
+		RequestsWithoutAuthHeader: requestsWithoutAuthHeader,
+		RequestsWithAuthHeader:    requestsWithAuthHeader,
 	}
 }
 
 func (mr *metricRegistry) getMiddleware() echo.MiddlewareFunc {
 	return func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) error {
-			startTime := time.Now()
+			isCpeRedirectRequest := strings.Contains(c.Request().RequestURI, "api")
```
Comment (Sachin4403) [inline metrics.go]: read "api" from configuration.

File: forwarder.go (hunk 1 above)
Comment (chaitanyasingla-dt) [reply forwarder.go]: Done.

File: metrics.go (hunk 3 above)
Comment (chaitanyasingla-dt) [reply metrics.go]: Done.

File: petasos-rewriter.yaml (hunk 2 above)
Comment (chaitanyasingla-dt) [reply petasos-rewriter.yaml]: Done.

**Review by Sachin4403 — APPROVED** [review]

**Pull request merged**